src/EditorFeatures/Core/Diagnostics/AbstractPushOrPullDiagnosticsTaggerProvider.SingleDiagnosticKindPullTaggerProvider.cs

```diff
@@ -155,6 +158,25 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView? textVi
             }
         }
 
+        private bool HasSuccessfullyLoaded(Project? project, HashSet<Project> seenProjects)
+        {
+            if (project != null && seenProjects.Add(project))
```

Comment:
note: the set is mostly paranoia on my part.  i don't think we support circular project references, but i don't want to risk it.